relay-server/src/endpoints/common.rs

```diff
@@ -405,134 +405,147 @@ where
 
     let future = project_manager
         .send(GetProject { public_key })
         .map_err(BadStoreRequest::ScheduleFailed)
-        .and_then(clone!(event_id, envelope_summary, scoping, |project| {
-            extract_envelope(&request, meta)
-                .into_future()
-                .and_then(clone!(event_id, envelope_summary, |envelope| {
-                    event_id.replace(envelope.event_id());
-                    envelope_summary.replace(EnvelopeSummary::compute(&envelope));
-
-                    if envelope.is_empty() {
-                        Err(BadStoreRequest::EmptyEnvelope)
-                    } else {
-                        Ok(envelope)
-                    }
-                }))
-                .and_then(clone!(project, |envelope| {
-                    project
-                        .send(CheckEnvelope::cached(envelope))
-                        .map_err(BadStoreRequest::ScheduleFailed)
-                        .and_then(|result| result.map_err(BadStoreRequest::ProjectFailed))
-                }))
-                .and_then(clone!(scoping, |response| {
-                    scoping.replace(response.scoping);
-
-                    let checked = response.result.map_err(BadStoreRequest::EventRejected)?;
-
-                    // Skip over queuing and issue a rate limit right away
-                    let envelope = match checked.envelope {
-                        Some(envelope) => envelope,
-                        None => return Err(BadStoreRequest::RateLimited(checked.rate_limits)),
-                    };
-                    // TODO: Update envelope_summary from checked.envelope, once the rate-limiting
-                    // code in CheckEnvelope emits its own outcomes.
-
-                    if check_envelope_size_limits(&config, &envelope) {
-                        Ok((envelope, checked.rate_limits))
-                    } else {
-                        Err(BadStoreRequest::PayloadError(StorePayloadError::Overflow))
-                    }
-                }))
-                .and_then(clone!(event_manager, project_manager, |(
-                    envelope,
-                    rate_limits,
-                )| {
-                    type RetVal = ResponseFuture<
-                        (Envelope, RateLimits, Option<Addr<Project>>),
-                        BadStoreRequest,
-                    >;
-                    // do dynamic sampling on transactions
-                    let trace_context = match envelope.trace_context() {
-                        None => {
-                            // if we don't have a trace context we can't do dynamic sampling so stop.
-                            return Box::new(Ok((envelope, rate_limits, None)).into_future())
-                                as RetVal;
+        .and_then(clone!(
+            event_id,
+            envelope_summary,
+            scoping,
+            outcome_producer,
+            |project| {
```

Comment:
😱 Rustfmt sure had some opinions about this change. "Hide whitespace changes" recommended, I guess.